### PR TITLE
WIP: DO NOT MERGE: RHOARDOC-1133

### PR DESCRIPTION
**Pull request by inoxx03**
Conditionalized debugging section, to be shared by all of the Java-based runtimes.
AsciiDoctor Build is currently broken. do not merge.

### CODE REVIEW

**Comment by rhoads-zach** [comment]
@inoxx03 LGTM, merging

File: docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+* When launching the application manually using the `mvn vertx:debug` goal. This starts the application with debugging enabled.
+endif::[]
+ifdef::spring-boot[]
+* When launching the application manually using the `mvn spring-boot:run`goal.
+endif::[]
+ifdef::wf-swarm[]
+* When starting the application without waiting for it to exit using the `mvn wildfly-swarm:start` goal.
 This is useful especially when performing integration testing.
+endif::[]
 * When using the Arquillian adapter for {runtime}.
```
Comment (Ladicek) [inline docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc]: Only WildFly Swarm has an Arquillian adapter.

File: docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc [hunk 2]
```diff
@@ -17,15 +26,36 @@ This is useful especially when performing integration testing.
...
 
 . In a console, navigate to the directory with your application.
+ifdef::wf-swarm[]
 . Launch your application and specify the debug port using the `-Dswarm.debug.port` argument:
+endif::[]
+ifdef::vertx[]
+. Launch your application and specify the debug port using the `-Ddebug.port` argument:
+endif::[]
+ifdef::spring-boot[]
+. Launch your application and specify the debug port using the `-Dspring-boot.debug.port` argument:
```
Comment (Ladicek) [inline docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc]: Did you verify that these system properties are actually supported by the respective runtimes?

File: docs/vertx-runtime/master.adoc [hunk 3]
```diff
@@ -25,6 +26,27 @@ include::topics/vertx-community-and-product-versions.adoc[leveloffset=+2]
 
 include::topics/configuring-vertx.adoc[leveloffset=+2]
 
+== Debugging
+
+This sections contains information about debugging your {WildFlySwarm}&#x2013;based application both in local and remote deployments.
```
Comment (Ladicek) [inline docs/vertx-runtime/master.adoc]: Not really `{WildFlySwarm}` :-)

File: docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc (hunk 1 above)
Comment (Ladicek) [reply docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc]: Please ignore, commented about this in #1005.

File: docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc (hunk 2 above)
Comment (Ladicek) [reply docs/topics/proc_starting-your-application-locally-in-debugging-mode.adoc]: Please ignore, addressed in #1005.

File: docs/vertx-runtime/master.adoc (hunk 3 above)
Comment (Ladicek) [reply docs/vertx-runtime/master.adoc]: Please ignore, commented about this in #1005.

**Pull request merged**